src/plugins/NEMD/MettDeamon.cpp

```diff
@@ -0,0 +1,2048 @@
+
+	// calc drift again
+	sum_vxi = std::accumulate(vxi.begin(), vxi.end(), 0.0);
+	sum_vyi = std::accumulate(vyi.begin(), vyi.end(), 0.0);
+	sum_vzi = std::accumulate(vzi.begin(), vzi.end(), 0.0);
+
+	vxd = sum_vxi * dInvNumSamples;
+	vyd = sum_vyi * dInvNumSamples;
+	vzd = sum_vzi * dInvNumSamples;
+
+	cout << "vxd,vyd,vzd = " << vxd << "," << vyd << "," << vzd << endl;
+}
+
+MettDeamon::MettDeamon() :
+		_reservoir(nullptr),
+		_bIsRestart(false),
+		_bInitFeedrateLog(false),
+		_dAreaXZ(0.),
+		_dInvDensityArea(0.),
+		_dDeletedMolsPerTimestep(0.),
+		_dInvNumTimestepsSummation(0.),
+		_dMoleculeDiameter(1.0),
+		_dTransitionPlanePosY(0.0),
+		_dDensityTarget(0.0),
+		_dVolumeCV(0.0),
+		_nUpdateFreq(0),
+		_nWriteFreqRestart(0),
+		_nMaxReservoirMoleculeID(0),
+		_nNumMoleculesDeletedGlobalAlltime(0),
+		_nMovingDirection(MD_UNKNOWN),
+		_nFeedRateMethod(FRM_UNKNOWN),
+		_nZone2Method(Z2M_UNKNOWN),
+		_nNumValsSummation(0),
+		_numDeletedMolsSum(0),
+		_nDeleteNonVolatile(0)
+{
+	_dAreaXZ = global_simulation->getDomain()->getGlobalLength(0) * global_simulation->getDomain()->getGlobalLength(2);
+
+	// init restart file
+	std::stringstream fnamestream;
+	fnamestream << "MettDeamonRestart_movdir-" << (uint32_t)_nMovingDirection << ".dat";
+	std::ofstream ofs(fnamestream.str().c_str(), std::ios::out);
+	std::stringstream outputstream;
+	outputstream << "     simstep" << "   slabIndex" << "                  deltaY" << std::endl;
+	ofs << outputstream.str();
+	ofs.close();
+
+	// summation of deleted molecules
+	_listDeletedMolecules.clear();
+	_listDeletedMolecules.push_back(0);
+
+	// init identity change vector
+	uint8_t nNumComponents = global_simulation->getEnsemble()->getComponents()->size();
+	_vecChangeCompIDsFreeze.resize(nNumComponents);
+	_vecChangeCompIDsUnfreeze.resize(nNumComponents);
+	std::iota (std::begin(_vecChangeCompIDsFreeze), std::end(_vecChangeCompIDsFreeze), 0);
+	std::iota (std::begin(_vecChangeCompIDsUnfreeze), std::end(_vecChangeCompIDsUnfreeze), 0);
+
+	// density values
+	_vecDensityValues.clear();
+
+	// particle reservoir
+	_reservoir = new Reservoir(this);
+
+	// init manipulation count for determination of the feed rate
+	_feedrate.numMolecules.inserted.local = 0;
+	_feedrate.numMolecules.deleted.local = 0;
+	_feedrate.numMolecules.changed_to.local = 0;
+	_feedrate.numMolecules.changed_from.local = 0;
+
+	// init feed rate sum
+	_feedrate.feed.sum = 0;
+
+	// init released counter
+	_released.count.local = 0;
+	_released.deleted.local = 0;
+	_released.init_file = false;
+
+//	// init colleague ptr
+//	_colleague.ptr = nullptr;
+
+	// seed rand()
+	srand (static_cast <unsigned> (time(0)));
+
+	// init released velocities vector
+	_released.log_v.clear();
+	_released.log_v.resize(0);
+	_released.init_file_vel = false;
+	_released.log_freq_vel = 5000;
+
+	// random numbers
+	DomainDecompBase& domainDecomp = global_simulation->domainDecomposition();
+	int nRank = domainDecomp.getRank();
+	_rnd = new Random(8624+nRank);
```

Comment:
not released -> change to std::unique_ptr!

Reply:
done